src/unstructured_client/_hooks/custom/split_pdf_hook.py

```diff
@@ -0,0 +1,505 @@
+
+    def before_request(
+        self, hook_ctx: BeforeRequestContext, request: requests.PreparedRequest
+    ) -> Union[requests.PreparedRequest, Exception]:
+        """If `splitPdfPage` is set to `true` in the request, the PDF file is split into
+        separate pages. Each page is sent as a separate request in parallel. The last
+        page request is returned by this method. It will return the original request
+        when: `splitPdfPage` is set to `false`, the file is not a PDF, or the HTTP
+        has not been initialized.
+
+        Args:
+            hook_ctx (BeforeRequestContext): The hook context containing information about
+            the operation.
+            request (requests.PreparedRequest): The request object.
+
+        Returns:
+            Union[requests.PreparedRequest, Exception]: If `splitPdfPage` is set to `true`,
+            the last page request; otherwise, the original request.
+        """
+        operation_id = hook_ctx.operation_id
+        content_type = request.headers.get("Content-Type")
+        body = request.body
+        if not isinstance(body, bytes) or content_type is None:
+            return request
+
+        decoded_body = MultipartDecoder(body, content_type)
+        form_data = self._parse_form_data(decoded_body)
+        split_pdf_page = form_data.get(PARTITION_FORM_SPLIT_PDF_PAGE_KEY)
+        if split_pdf_page is None or split_pdf_page == "false":
+            return request
+
+        file = form_data.get(PARTITION_FORM_FILES_KEY)
+        if (
+            file is None
+            or not isinstance(file, File)
+            or not file.filename.endswith(".pdf")
+        ):
+            logger.warning("Given file is not a PDF. Continuing without splitting.")
+            return request
+
+        if self.client is None:
+            logger.warning("HTTP client not accessible! Continuing without splitting.")
+            return request
+
+        filename = file.filename.replace(".pdf", "")
+        pages = self._get_pdf_pages(file.content)
+        call_api_partial = functools.partial(
+            self._call_api,
+            request=request,
+            form_data=form_data,
+            filename=filename,
+        )
+        call_threads = self._get_split_pdf_call_threads()
+        self.partition_requests[operation_id] = []
+        with ThreadPoolExecutor(max_workers=call_threads) as executor:
+            for page_content, page_number, all_pages_number in pages:
+                self.partition_requests[operation_id].append(
+                    executor.submit(call_api_partial, (page_content, page_number))
+                )
+                if page_number == all_pages_number - 1:
+                    break
+
+        last_page_request = self._create_request(
+            request, form_data, next(pages)[:2], filename
+        )
+        last_page_prepared_request = self.client.prepare_request(last_page_request)
+        return last_page_prepared_request
```

Comment:
I assume the division of requests to have the last page sent separately, is needed to make it work with the hooks way of work, because it needs to return some request. It would be good to point it out explicitely for reader in comment.

Reply:
Also, the last page is not put into `self.partition_requests` - I assume it's ok? The _await_elements doesn't use it? Couldn't see in code where it is picked up.

Reply:
It's passed to `_await_elements` as a response argument in `after_success` or `after_error`:

at the bottom of the `_await_elements` function 
```python
if response.status_code == 200:
    elements.append(response.json())
```

Reply:
Added comment here and to the `after_success` and `after_error` methods